docs/core/geometry.md

```diff
@@ -209,41 +223,57 @@ Possible extensions are:
  * `HomogeneousSurfaceMaterial`, homogeneous material description on a surface
  * `BinnedSurfaceMaterial`, an arbitrarily binned material description with a
     corresponding `BinUtility` object
+
+In addition, a dedicated extension exists to allow configuration of the material
+mapping process, that is in further described below.
+
  * `ProtoSurfaceMaterial`, only binning description (without material) to be
    used in the material mapping process
 
-# Geometry building
+## Geometry building
 
 The geometry building procedure follows the ATLAS TrackingGeometry philosophy of
 a static frame of *glued* volumes, that lead the navigation flow through the
 geometry,
 
-## Attaching a 3D detector geometry
+### Attaching a 3D detector geometry
 
 Usually, a 3D detector model geometry exists, which is either native to the full
 detector simulation (Geant4) or is translated into it. This model, however, is
 in general too detailed for track reconstruction: navigating through the
-detailed detector geometry
+detailed detector geometry is generally costly and one can profit greatly from a simplication mechanism.
 
 For most part of the track reconstruction, only a surface based description of
 the detector is needed, in order to allow (surface based) material integration
-and parametrization/ prediction of trajectories on detection surfaces. It is thus
+and parametrization/prediction of trajectories on detection surfaces. It is thus
 necessary that the detection surfaces are described to full detail in the
 reconstruction geometry (called `TrackingGeometry`). This is guaranteed by a
 proxy mechanism that connects the detection elements (conveniently called
 `DetectorElement`) to `Surface` object in the reconstruction:
 
 ![DetectorElement](/figures/geometry/DetectorElement.png)
 
-### Existing plugins for 3D geometry libraries
+#### Existing plugins for 3D geometry libraries
 
 Very simple helper methods for 3D libraries exist, they are certainly not
 optimised, but used for templating:
 
-* `TGeoDetectorElement` connects a TGeo volume to a `Surface` (see [](TGeo plugin))
-* `DD4HepDetectorElement` connects a DD4hep volume (based on TGeo) to a `Surface` (see [](DD4hep plugin))
+* `TGeoDetectorElement` connects a TGeo volume to a `Surface`
+* `DD4HepDetectorElement` connects a DD4hep volume (based on TGeo) to a `Surface`
```

Comment:
We could add the `Geant4DetectorElement` here as well, maybe also with a link to the documentation (via the `{class}` prefix and namespace)

Reply:
This will have to be updated anyway for the `Detector` infrastructure, I will make a new PR once this is done.